setup.py

```diff
@@ -116,13 +117,59 @@ def get_setup_kwargs(raw=False):
     return kwargs
 
 
+def avoid_common_setup_errors(raw=False):
+    # This runs before the setup, trying to catch common setup errors
+    try:
+        get_distribution("telegram")
+        print(
+            """
+            =============================DEBUG ASSISTANCE==========================
+            You can not use python-telegram-bot and telegram. Please run 
+            pip uninstall telegram and try again.
```

Comment:
The first sentence is a bit opaque for people that are not pip/python seniors IMO. How about

```suggestion
            You are trying to install the `python-telegram-bot` library, but the `telegram` library is already installed. Both libraries provide a Python package called `telegram`, which will lead to a namespace clash.
            If you want to use the `python-telegram-bot` library, please first uninstall the `telegram` library before installing the `python-telegram-bot` library again.
```

This also makes no assumptions on how the user (un)installs packages in their environment … Similar below.